test/service/test_request_validation.py

```diff
@@ -177,3 +181,55 @@ def test_manifest(self):
             manifest_config = json.load(open('{}/request_config.json'.format(self.service_config_dir), 'r'))['manifest']
             expected_fieldnames = list(manifest_config['bundles'].keys()) + list(manifest_config['contents.files'].keys())
             self.assertEqual(expected_fieldnames, tsv_file.fieldnames, 'Manifest headers are not configured correctly')
+
+    @mock_sts
+    @mock_s3
+    def test_bdbag_manifest(self):
+        """
+        As in test_manifest, moto will mock the requests.get call so we can't hit localhost; add_passthru let's us hit
+        the server (see GitHub issue and comment: https://github.com/spulec/moto/issues/1026#issuecomment-380054270)
+        """
+        logging.getLogger('test_request_validation').warning('test_manifest is invoked')
+        with ResponsesHelper() as helper, TemporaryDirectory() as zip_dir:
+            helper.add_passthru(self.base_url)
+            storage_service = StorageService()
+            storage_service.create_bucket()
+            url = self.base_url + '/repository/files/export?filters={"file":{}}&format=bdbag'
+            response = requests.get(url)
+            self.assertEqual(200, response.status_code, 'Unable to download manifest')
+            with ZipFile(response.text, 'r') as zip_fh:
+                zip_fh.extractall(zip_dir)
+            zip_fname = os.path.basename(os.path.splitext(response.text)[0])
+            with open(os.path.join(zip_dir, zip_fname, 'data', 'participant.tsv'), 'r') as fh:
+                participant = csv.reader(fh, delimiter='\t')
+                expected = ['entity:participant_id', '7b07b9d0-cc0e-4098-9f64-f4a569f7d746']
+                observed = []
+                for row in participant:
+                    observed.append(row[0])
+                self.assertEqual(len(observed), 2, 'participant.tsv contains duplicates.')
+                def checkListEqual(L1, L2):
+                    return len(L1) == len(L2) and sorted(L1) == sorted(L2)
+                self.assertTrue(checkListEqual(expected, observed), 'participant.tsv contains incorrect data')
+
+            with open(os.path.join(zip_dir, zip_fname, 'data', 'sample.tsv'), 'r') as fh:
+                expected_fieldnames = ['entity:sample_id', 'participant_id', 'cell_suspension_id', 'bundle_uuid',
```

Comment:
It would help to correlate the column names with the values and to list each field on a separate line.

```suggestion
                expectations = [
                    ('entity:sample_id', 'DID_scRSq06'),
                    ...
                ]
                   
expected_header, expected_row = zip(*expectations)
```

In this case I am fine with hanging indent, BTW.

Reply:
You may also be able to inline `file_url` in that case.

Reply:
Implemented, looks cleaner and easier to read now.